core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -82,6 +77,11 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   // List of application logs to be deleted by event log cleaner.
   private var attemptsToClean = new mutable.ListBuffer[FsApplicationAttemptInfo]
 
+  // Mapping of log file names to their last modified time. This is used to ignore logs that
+  // are older during subsequent scans, to avoid processing data that is already known.
+  @volatile private var lastModifiedTimes: mutable.LinkedHashMap[String, Long]
```

Comment:
Thanks for your review. I have add the logical for add entries to `lastModifiedTimes`. Any suggestion for the type of `lastModifiedTimes`?